packages/ipfs/src/core/components/pin/add.js

```diff
@@ -66,9 +93,11 @@ module.exports = ({ pinManager, gcLock, dag }) => {
     const release = await gcLock.readLock()
 
     try {
-      await pinAdd()
+      return await pinAdd()
```

Comment:
I think this `return` here is intended behavior, given that on line 90 there's `return pinAdd()`. At least that is what TS identifying as type mismatch.